app/src/main/java/com/hillnerds/soundsparrow/ChannelSoundGeneration.java

```diff
@@ -0,0 +1,177 @@
+package com.hillnerds.soundsparrow;
+
+import android.util.Log;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Random;
+
+/**
+ * Created by aga on 06/02/17.
+ */
+
+public class ChannelSoundGeneration {
+
+    int scale;
+    int[] steps;
+
+    //a hard coded list of all possible instruments
+    private static Instrument[] instrumentList = new Instrument[] {
+            new Instrument("piano", 1, 40, 60),
+            new Instrument("guitar", 26, 45, 70),
+            new Instrument("trombone", 58, 55, 70),
+            new Instrument("trumpet", 57, 64, 80),
+            new Instrument("violin", 41, 50, 65),
+            new Instrument("saxophone", 66, 45, 60),
+            new Instrument("flute", 74, 73, 90)
+    };
+
+    /**
+     * This function chooses a rondom instrument form a hardcoded instrumentList.
+     * @param random - a random generator Object
+     * @return - return an Instrument object from instrumentList
+     */
+    public static Instrument generateRandomInstrument(Random random){
+        int instrumentNumber = random.nextInt((6 - 0) + 1) + 0;
+        return instrumentList[instrumentNumber];
+    }
+
+    /**
+     * Generate a MIDI file for a single Channel.
+     * @param c - A single channel object
+     * @return - An array of MidiSequence Objects
+     */
+    public static ArrayList<MidiSequence> generateMidiChannel(Channel c) {
+        ArrayList<MidiSequence> midiSequenceArray = new ArrayList<>();
+
+        int timestampCounter = 0;
+
+        StartingSequence instrument_change = new StartingSequence(192 + c.number, c.instrument, timestampCounter);
+        midiSequenceArray.add(instrument_change);
+
+        /**
+         * TODO: implement the way of setting the length of the track differently (ie. a number of
+         * bars so that tunes are synchronised)
+         */
+        for (int i = 0; i < 16; i = i+2) {
```

Comment:
Why i+2 when i isn't being used in the loop, could this be simplified? Also, ideally i should be named more descriptively.